python/ray/workflow/workflow_context.py

```diff
@@ -9,32 +9,60 @@ class WorkflowStepContext:
     def __init__(self,
```

Comment:
Could we make it a data class and have a function to set them? It can make it clear what are in the context. We also need a detailed description of what are each field does.